components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -49,6 +50,13 @@ internal class EngineObserver(
         }
     }
 
+    private fun isHostEquals(sessionUrl: String, newUrl: String): Boolean {
+        val sessionUri = URI(sessionUrl)
+        val newUri = URI(newUrl)
+
+        return sessionUri.scheme == newUri.scheme && sessionUri.host == newUri.host
```

Comment:
nit: There's an `toUri()`extension method in core-ktx afaik - which will use `Uri` that seems to be more forgiving than `URI`.

Reply:
The deal with this extension method is that it calls the Uri.parse() method of the android.net package. So, many of our unit tests would have to be changed to run as instrumented unit tests. Just wanted to check with you before making that change. Would probably require us to strip some methods from the unit test file and move them to a new instrumented test file running under the AndroidJunit4 test runner. Could end up being a more involved change.

Reply:
Using the other runner is enough; they do not need to become instrumented tests running on devices/emulators. We run those unit tests with Robolectric locally. They only reason why we avoid that runner if possible is that it is a bit slower to startup (a couple of seconds) compared to plain unit tests.